modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/resource/v1_0/ObjectEntryRelatedObjectsResourceImpl.java

```diff
@@ -65,14 +70,32 @@ public void deleteCurrentObjectEntry(
 				_objectDefinition.getStorageType());
 
 		_checkCurrentObjectEntry(objectEntryManager, currentObjectEntryId);
-		_checkRelatedObjectEntry(
-			objectEntryManager, objectRelationshipName, relatedObjectEntryId);
 
 		ObjectRelationship objectRelationship =
 			_objectRelationshipService.getObjectRelationship(
 				_objectDefinition.getObjectDefinitionId(),
 				objectRelationshipName);
 
+		ObjectDefinition relatedObjectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectRelationship.getObjectDefinitionId2());
+
+		if (relatedObjectDefinition.isSystem()) {
+			if (!GetterUtil.getBoolean(
+					PropsUtil.get("feature.flag.LPS-162966"))) {
+
+				throw new NotFoundException();
+			}
+
+			_checkSystemObjectEntry(
+				relatedObjectEntryId, relatedObjectDefinition);
+		}
+		else {
+			_checkRelatedObjectEntry(
+				objectEntryManager, objectRelationshipName,
+				relatedObjectEntryId);
+		}
+
 		ObjectRelatedModelsProvider objectRelatedModelsProvider =
 			_objectRelatedModelsProviderRegistry.getObjectRelatedModelsProvider(
 				_objectDefinition.getClassName(), objectRelationship.getType());
```

Comment:
We can disccus it tomorrow in the grow plan meeting

Reply:
> ObjectRelatedModelsProvider manages relationships with related system objects? 
Yes. There are two classes that handle related system objects:
* SystemObject1toMObjectRelatedModelsProviderImpl
* SystemObjectMtoMObjectRelatedModelsProviderImpl
When you provide the related object definition to the `_objectRelatedModelsProviderRegistry` it will return the proper implementation of the `ObjectRelatedModelsProvider`. As you provide the object definition and the relationship type, the registry will know if it is a system or a custom and the type of the relatioship. In this case, as the related object definition is a system and the relationship type is a one-to-many, it will return `SystemObject1toMObjectRelatedModelsProviderImpl`

This is from the class `SystemObject1toMObjectRelatedModelsProviderImpl`
```
@Override
	public void disassociateRelatedModels(
			long userId, long objectRelationshipId, long primaryKey1,
			long primaryKey2)
		throws PortalException {

		ObjectRelationship objectRelationship =
			_objectRelationshipLocalService.getObjectRelationship(
				objectRelationshipId);

		_objectEntryLocalService.insertIntoOrUpdateExtensionTable(
			objectRelationship.getObjectDefinitionId2(),
			GetterUtil.getLong(primaryKey2),
			HashMapBuilder.<String, Serializable>put(
				() -> {
					ObjectField objectField =
						_objectFieldLocalService.getObjectField(
							objectRelationship.getObjectFieldId2());

					return objectField.getName();
				},
				0
			).build());
	}
```

As you can see, it is not using the `ObjectEntryLocalService#updateObjectEntry` method (it is using the `ObjectEntryLocalService` for other purposes 😅 )

Also, this is the implementation used in the UI, if you delete a related system object, the code used is this one.

My unique concern is about the use of local services instead of remote services. I know that we talked with the object team to change this and they did it in the custom objects, maybe we can talk to them or we can change it and send to them to review it.

Reply:
As we discuss today, it was strange that I've tried it yesterday and it never received the `SystemObject1toMObjectRelatedModelsProviderImpl` implementation :sweat_smile: It could be very nice to discuss with objects team the possibility of insert/implement checking permission when we manage relationships and make sure that these methods can be supported for the remote service